kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -170,16 +170,21 @@ private static InformerEventSource<KafkaService, KafkaProxy> buildKafkaServiceEv
 
     @VisibleForTesting
     static SecondaryToPrimaryMapper<KafkaService> kafkaServiceRefToProxyMapper(EventSourceContext<KafkaProxy> context) {
-        return kafkaServiceRef -> {
+        return kafkaService -> {
+            // we do not want to trigger reconciliation of any proxy if the ingress has not been reconciled
+            if (!ResourcesUtil.isStatusFresh(kafkaService)) {
+                LOGGER.debug("Ignoring event from KafkaService with stale status: {}", ResourcesUtil.toLocalRef(kafkaService));
+                return Set.of();
+            }
```

Comment:
Feels like there's nearly a decorator we could have here. But it's slightly ugly because of the lack of a common status class -- we'd need to pass the observedGeneration extractor function, so it's probably not worth saving the 3 lines × #CRs.